version: '3.8'

volumes:
  userdata:
    name: external_vol-domoticz-userdata
    external: true

networks:
  domoticz:

services:
  domoticz:
    image: domoticz/domoticz:2021-beta.13738
    env_file: external_env-all.env
    # environment:
    #   - LOG_PATH=/opt/domoticz/userdata/domoticz.log
    volumes:
      - userdata:/opt/domoticz/userdata
    networks:
{% if VARS.NETWORKS | type_debug == "dict" %}
      domoticz:
{% else %}
      - domoticz
{% endif %}
{% if VARS.NETWORKS | length > 0 %}
      {{ VARS.NETWORKS | to_nice_yaml(indent=2) | trim | indent(6) }}
{% endif %}
{% if VARS.PORTS | length > 0 %}
    ports:
      # - 8080:8080
      {{ VARS.PORTS | to_nice_yaml(indent=2) | trim | indent(6) }}
{% endif %}
{% if VARS.DEVICES | length > 0 %}
    devices:
      {{ VARS.DEVICES | to_nice_yaml(indent=2) | trim | indent(6) }}
{% endif %}
    # network_mode: "host"
    container_name: domoticz
    restart: always
    healthcheck:
      test: curl -f http://localhost:8080 || exit 1
      interval: 60s
      timeout: 5s
      retries: 10
      start_period: 10m
